runtime/syntax/generator/vim.vim.base

```diff
@@ -862,218 +863,266 @@ syn region	vimHiLink	contained matchgroup=Type start="\%(\<hi\%[ghlight]!\=\s\+\
-VimL syn region	vimLineComment	start=+^[ \t:]*\zs".*$+ skip=+\n\s*\\\|\n\s*"\\ + end="$" contains=@vimCommentGroup,vimCommentString,vimCommentTitle
-
-syn match	vimCommentTitle	'"\s*\%([sS]:\|\h\w*#\)\=\u\w*\(\s\+\u\w*\)*:'hs=s+1	contained contains=vimCommentTitleLeader,vimTodo,@vimCommentGroup
-syn match	vim9CommentTitle	'#\s*\%([sS]:\|\h\w*#\)\=\u\w*\(\s\+\u\w*\)*:'hs=s+1	contained contains=vim9CommentTitleLeader,vimTodo,@vimCommentGroup
-
-" allowed anywhere in the file
-if !s:vim9script
-  syn match	vimShebangError	"^\s*\zs#!.*" display
-endif
-syn match	vimShebang	"\%^#!.*" display
-
-syn match	vimContinue		"^\s*\zs\\"
-syn match         vimContinueComment	'^\s*\zs["#]\\ .*'
-syn cluster	vimContinue contains=vimContinue,vimContinueComment
-syn region	vimString	start="^\s*\\\z(['"]\)" skip='\\\\\|\\\z1' end="\z1" oneline keepend contains=@vimStringGroup,vimContinue
-syn match	vimCommentTitleLeader	'"\s\+'ms=s+1	contained
-syn match	vim9CommentTitleLeader	'#\s\+'ms=s+1	contained
-
-" Searches And Globals: {{{2
-" ====================
-syn match	vimSearch	'^\s*[/?].*'		contains=vimSearchDelim
-syn match	vimSearchDelim	'^\s*\zs[/?]\|[/?]$'	contained
-syn region	vimGlobal	matchgroup=Statement start='\<g\%[lobal]!\=/'  skip='\\.' end='/'	skipwhite nextgroup=vimSubst1
-syn region	vimGlobal	matchgroup=Statement start='\<v\%[global]!\=/' skip='\\.' end='/'	skipwhite nextgroup=vimSubst1
-
-" Vim9 Script Regions: {{{2
-" ==================
-
-if s:vim9script
-  syn cluster vimLegacyTop	contains=TOP,vim9LegacyHeader,vim9Comment,vim9LineComment
-  VimFoldH syn region vim9LegacyHeader start="\%^" end="^\ze\s*vim9s\%[cript]\>" contains=@vimLegacyTop,vimComment,vimLineComment
-
-  syn keyword vim9Vim9ScriptArg	noclear contained
-  syn keyword vim9Vim9Script	vim9s[cript] nextgroup=vim9Vim9ScriptArg skipwhite
-endif
-
 " Embedded Scripts:  {{{2
 " ================
-"   perl,ruby     : Benoit Cerrina
-"   python,tcl    : Johannes Zellner
+"   perl,ruby	: Benoit Cerrina
+"   python,tcl	: Johannes Zellner
 "   mzscheme, lua : Charles Campbell
 
 " Allows users to specify the type of embedded script highlighting
-" they want:  (perl/python/ruby/tcl support)
+" they want:  (lua/mzscheme/perl/python/ruby/tcl support)
 "   g:vimsyn_embed == 0   : don't embed any scripts
-"   g:vimsyn_embed =~# 'l' : embed Lua      (but only if vim supports it)
-"   g:vimsyn_embed =~# 'm' : embed MzScheme (but only if vim supports it)
-"   g:vimsyn_embed =~# 'p' : embed Perl     (but only if vim supports it)
-"   g:vimsyn_embed =~# 'P' : embed Python   (but only if vim supports it)
-"   g:vimsyn_embed =~# 'r' : embed Ruby     (but only if vim supports it)
-"   g:vimsyn_embed =~# 't' : embed Tcl      (but only if vim supports it)
+"   g:vimsyn_embed =~# 'l' : embed Lua
+"   g:vimsyn_embed =~# 'm' : embed MzScheme
+"   g:vimsyn_embed =~# 'p' : embed Perl
+"   g:vimsyn_embed =~# 'P' : embed Python
+"   g:vimsyn_embed =~# 'r' : embed Ruby
+"   g:vimsyn_embed =~# 't' : embed Tcl
```

Comment:
I'm wondering if we need to check if vim support the script.
I sometimes want to enable the highlight even if the current vim doesn't support it.

Reply:
Ah, I misunderstood your change.
Looks good.